projects/observability/src/shared/components/cartesian/d3/data/cartesian-data.ts

```diff
@@ -4,35 +4,40 @@ import {
   MouseLocationData,
   RelativeMouseLocation
 } from '../../../utils/mouse-tracking/mouse-tracking';
-import { AxisType, Series } from '../../chart';
+import { AxisType } from '../../chart';
 import { ChartTooltipTrackingOptions as ChartTooltipTrackingStrategy } from '../../chart-interactivty';
-import { SingleAxisDataLookupStrategy } from '../interactivity/data-strategy/single-axis-data-lookup-strategy';
 import { AnyCartesianScale, CartesianScaleBuilder } from '../scale/cartesian-scale-builder';
 
-export abstract class CartesianSeries<TData> {
-  protected dataLookupStrategy?: MouseDataLookupStrategy<TData, Series<TData>>;
+export abstract class CartesianData<TData, TVisualization> {
+  protected dataLookupStrategy?: MouseDataLookupStrategy<TData, TVisualization>;
   protected xScale: AnyCartesianScale<TData>;
   protected yScale: AnyCartesianScale<TData>;
 
-  public constructor(
-    protected readonly series: Series<TData>,
+  protected constructor(
+    protected readonly visualization: TVisualization,
     protected readonly scaleBuilder: CartesianScaleBuilder<TData>,
     tooltipTrackingStrategy?: ChartTooltipTrackingStrategy
   ) {
     this.xScale = this.buildXScale();
     this.yScale = this.buildYScale();
     if (tooltipTrackingStrategy) {
-      this.dataLookupStrategy = this.buildDataLookupStrategy(tooltipTrackingStrategy);
+      this.dataLookupStrategy = this.buildDataLookupStrategy(visualization, tooltipTrackingStrategy);
     }
   }
 
   public abstract drawSvg(element: BaseType): void;
   public abstract drawCanvas(context: CanvasRenderingContext2D): void;
+
   protected abstract buildMultiAxisDataLookupStrategy(
     tooltipStrategy: ChartTooltipTrackingStrategy
-  ): MouseDataLookupStrategy<TData, Series<TData>>;
+  ): MouseDataLookupStrategy<TData, TVisualization>;
+
+  protected abstract buildDataLookupStrategy(
+    visualization: TVisualization,
+    strategy: ChartTooltipTrackingStrategy
+  ): MouseDataLookupStrategy<TData, TVisualization>;
```

Comment:
Why remove the default implementation? Does every class already implement this?

Reply:
Hmm - I got confused because git shows it as a rename of series, which has many implementations. Looks like this is a separate class, and series is now showing up as an add. will re-review.